src/generic.rs

```diff
@@ -606,44 +627,50 @@ where
             )
             .await;
 
-            Python::attach(move |py| {
-                if cancelled(future_tx1.bind(py))
-                    .map_err(dump_err(py))
-                    .unwrap_or(false)
-                {
-                    return;
-                }
-
-                let _ = set_result(
-                    &locals2.event_loop(py),
-                    future_tx1.bind(py),
-                    result.and_then(|val| val.into_py_any(py)),
-                )
-                .map_err(dump_err(py));
-            });
-        })
-        .await
-        {
-            if e.is_panic() {
+            // We should not hold GIL inside async-std/tokio event loop,
+            // because a blocked task may prevent other tasks from progressing.
+            R::spawn_blocking(|| {
```

Comment:
Kinda hard to tell from the diff, but the only change here is the addition of `spawn_blocking`?